src/driver/alloy.rs

```diff
@@ -0,0 +1,348 @@
+        }
+
+        let bundle_filler = BundleBlockFiller::from(self.bundle.clone());
+
+        let run_result = trevm.try_with_block(&bundle_filler, |trevm| {
+            let mut trevm = trevm;
+
+            let txs = self
+                .bundle
+                .txs
+                .iter()
+                .map(|tx| TxEnvelope::decode_2718(&mut tx.chunk()))
+                .collect::<Result<Vec<_>, _>>();
+            let txs = match txs {
+                Ok(txs) => txs,
+                Err(e) => return Err(trevm.errored(BundleError::TransactionDecodingError(e))),
+            };
+
+            for tx in txs.iter() {
+                let run_result = trevm.run_tx(tx);
+
+                match run_result {
+                    // return immediately if errored
+                    Err(e) => {
+                        return Err(e.map_err(|e| BundleError::EVMError { inner: e }));
+                    }
+                    // Accept + accumulate state
+                    Ok(res) => match res.result() {
+                        ExecutionResult::Revert { .. } | ExecutionResult::Halt { .. } => {
+                            return Err(res.errored(BundleError::BundleReverted));
+                        }
+                        ExecutionResult::Success { .. } => {
+                            let (execution_result, committed_trevm) = res.accept();
+                            let block_env = committed_trevm.inner().block();
+
+                            let from_address = tx.recover_signer().map_err(|e| BundleError::TransactionSenderRecoveryError(e));
+                            let from_address = match from_address {
+                                Ok(addr) => addr,
+                                Err(e) => return Err(committed_trevm.errored(e)),
+                            };
+
+                            let mut tx_response = EthCallBundleTransactionResult::default();
+                            tx_response.tx_hash = *tx.tx_hash();
+                            tx_response.value = execution_result.output().cloned();
+                            tx_response.gas_used = execution_result.gas_used();
+                            tx_response.from_address = from_address;
+                            tx_response.to_address = match tx.to() {
+                                TxKind::Call(to) => Some(to),
+                                _ => Some(Address::ZERO),
+                            };
+                            tx_response.gas_price = match tx {
+                                TxEnvelope::Legacy(tx) => U256::from(tx.tx().gas_price),
+                                TxEnvelope::Eip2930(tx) => U256::from(tx.tx().gas_price),
+                                TxEnvelope::Eip1559(tx) => U256::from(tx.tx().effective_gas_price(Some(block_env.basefee.to::<u64>()))),
+                                TxEnvelope::Eip4844(_) => U256::ZERO,
+                                _ => panic!("Unsupported transaction type"),
```

Comment:
unless these are filtered somewhere i'm not seeing, this should be an error instead of a panic?